src/test/java/com/wikia/webdriver/testcases/discussions/SortingTests.java

```diff
@@ -23,54 +23,74 @@
   }
 
   @Execute(asUser = User.USER_3)
   @InBrowser(emulator = Emulator.DESKTOP_BREAKPOINT_BIG)
-  public void loggedUserOnDesktopCanSwitchBetweenLatestAndTrendingTab() {
-    userCanSwitchBetweenLatestAndTrendingTab();
+  public void userOnDesktopCanSwitchBetweenLatestAndTrendingSortOptions() {
+    userCanSwitchBetweenLatestAndTrendingOptions(new PostsListPage().open());
   }
 
-  private void userCanSwitchBetweenLatestAndTrendingInDropdown() {
-    PostsListPage page = new PostsListPage().open();
-    FiltersPopOver filtersPopOver = page.getFiltersPopOver();
-    DiscussionsHeader discussionsHeader = page.getDiscussionsHeader();
-    discussionsHeader.clickSortButtonOnMobile();
-
-    Assertion.assertTrue(filtersPopOver.isSortListVisibleMobile());
+  @Execute(asUser = User.USER_4)
+  @InBrowser(emulator = Emulator.DESKTOP_BREAKPOINT_BIG)
+  public void userOnDesktopCannotChangeSortingOnFollowingTab() {
+    Assertion.assertFalse(new FollowPage()
+      .open()
+      .getSortingFilterDesktop()
+      .isEnabled());
+  }
 
-    filtersPopOver.clickLatestLinkOnMobile();
-    filtersPopOver.clickApplyButton();
-    new Loading(driver).handleAsyncPageReload();
-    discussionsHeader.clickSortButtonOnMobile();
-    filtersPopOver.clickTrendingOptionInSortMenu();
-    filtersPopOver.clickApplyButton();
-    new Loading(driver).handleAsyncPageReload();
+  @Test(enabled = false)
+  @RelatedIssue(issueID = "IRIS-4995")
+  @Execute(asUser = User.USER_3)
+  @InBrowser(browser = Browser.CHROME, emulator = Emulator.GOOGLE_NEXUS_5)
+  public void userOnMobileCannotChangeSortingOnFollowingTab() {
+    Assertion.assertFalse(new FollowPage()
+      .open()
+      .getDiscussionsHeader()
+      .openFilterMenu()
+      .isSortingFilterEnabled());
   }
 
-  private void userCanSwitchBetweenLatestAndTrendingTab() {
-    SortingFiltersOnDesktop filters = new PostsListPage().open().getSortingFiltersOnDesktop();
-    filters.clickLatestOption();
-    new Loading(driver).handleAsyncPageReload();
+  private void userCanSwitchBetweenLatestAndTrendingInDropdown(PageWithPosts page) {
+    performSortingCheckOnMobile(page, SortOption.LATEST);
+    performSortingCheckOnMobile(page, SortOption.TRENDING);
+  }
 
-    Assertion.assertTrue(driver.getCurrentUrl().contains("latest"));
+  private void userCanSwitchBetweenLatestAndTrendingOptions(PageWithPosts page) {
+    performSortingCheckOnDesktop(page, SortOption.LATEST);
+    performSortingCheckOnDesktop(page, SortOption.TRENDING);
+  }
 
-    filters.clickTrendingOption();
-    new Loading(driver).handleAsyncPageReload();
+  private void performSortingCheckOnMobile(PageWithPosts page, SortOption sortOption) {
+    page
+      .getDiscussionsHeader()
+      .openFilterMenu()
+      .chooseSortingOption(sortOption)
+      .clickApplyButton()
+      .waitForPageReload();
+    Assertion.assertTrue(page.getCurrentUrl().contains(sortOption.getQuery()));
```

Comment:
is this sufficient check?

Reply:
As discussed offline, the only way we could safely perform checking if posts are sorted is to have a dedicated wiki with known state of posts. I can create a separate task to do it, but to tell the truth, I don't see much value. I'd rather make sure this is covered by tests on backend side.